packages/json-api/src/Resource.ts

```diff
@@ -0,0 +1,259 @@
+import {
+  JSONAPIResource,
+  JSONAPIDocument,
+  JSONObject,
+  JSONValue,
+  JSONAPIResourceRelationship,
+  JSONAPIResourceId
+} from "./jsonApiTypes";
+import {
+  IncludesMap,
+  MapObject,
+  Registry,
+  Resource,
+  ResourceAttributesSpec,
+  ResourceParseOptions,
+  ResourceRelationshipsSpec,
+  ResourceSpec
+} from "./types";
+
+type ResourceIdMapper = {
+  attr: string;
+  format(value: JSONValue): string | null | undefined;
+  parse(resource: JSONAPIResourceId): string | null;
+};
+
+type ResourceAttributeMapper = {
+  getter(data: JSONObject, key: string): any;
+  formatter(data: JSONValue): JSONValue;
+};
+
+type ResourceAttributeHelpers = Array<[string, ResourceAttributeMapper]>;
+
+const defaultIdMapper: ResourceIdMapper = {
+  attr: "id",
+  format(value) {
+    if (value === null) {
+      return null;
+    }
+    if (value === undefined) {
+      return undefined;
+    }
+    return String(value);
+  },
+  parse(resourceObj) {
+    return resourceObj.id != null ? String(resourceObj.id) : null;
```

Comment:
Not mixing ternary and if makes it easier to read.

```suggestion
    if (resourceObj.id === null) {
      return null
    }
    return String(resourceObj.id);
```

Reply:
OK, though it looks like overhead.